hw/ip/kmac/data/kmac.hjson

```diff
@@ -0,0 +1,340 @@
+      local:   "true"
+    }
+    { name: "NumWordsPrefix"
+      type: "int"
+      default: "11"
+      desc: "Number of words for Encoded NsPrefix."
+      local:"true"
+    }
+  ]
+  regwidth: "32"
+  registers: [
+    { name: "CFG"
+      desc: '''KMAC Configuration register.
+
+            This register is  updated when the hashing engine is in Idle.
+            If the software updates the register while the engine computes, the
+            updated value will be discarded.
+            '''
+      hwext:    "true"
+      hwqe:     "true"
+      swaccess: "rw"
+      hwaccess: "hrw"
+      fields: [
+        { bits: "0"
+          name: "kmac_en"
+          desc: '''KMAC datapath enable.
+
+                If this bit is 1, the incoming message is processed in KMAC
+                with the secret key.
+                '''
+          tags: [// don't enable kmac and sha data paths - we will do that in functional tests
+                 "excl:CsrNonInitTests:CsrExclWrite"]
+        } // f: kmac_en
+        { bits: "3:1"
+          name: "strength"
+          desc: '''Hashing Strength
+
+                3 bit field to select the security strength of SHA3 hashing
+                engine. If mode field is set to SHAKE or cSHAKE, only 128 and
+                256 strength can be selected. Other value will result error
+                when hashing starts.
+                '''
+          enum: [
+            { value: "0"
+              name:  "L128"
+              desc: "128 bit strength. Keccak rate is 1344 bit"
+            }
+            { value: "1"
+              name:  "L224"
+              desc: "224 bit strength. Keccak rate is 1152 bit"
+            }
+            { value: "2"
+              name:  "L256"
+              desc: "256 bit strength. Keccak rate is 1088 bit"
+            }
+            { value: "3"
+              name:  "L384"
+              desc: "384 bit strength. Keccak rate is 832 bit"
+            }
+            { value: "4"
+              name:  "L512"
+              desc: "512 bit strength. Keccak rate is 576 bit"
+            }
+          ]
+        } // f: strength
+        { bits: "5:4"
+          name: "mode"
+          desc: '''Keccak hashing mode.
+
+                This module supports SHA3 main hashing algorithm and the part
+                of its derived functions, SHAKE and cSHAKE with limitations.
+                This field is to select the mode.
+                '''
+          enum: [
+            { value: "0"
+              name: "SHA3"
+              desc: "SHA3 hashing mode. It appends `2'b 10` to the end of msg"
+            }
+            { value: "2"
+              name: "SHAKE"
+              desc: "SHAKE hashing mode. It appends `1111` to the end of msg"
+            }
+            { value: "3"
+              name: "cSHAKE"
+              desc: "cSHAKE hashing mode. It appends `00` to the end of msg"
+            }
+          ]
+        } // f: mode
```

Comment:
just curious, this `f: blah` term that shows up here, is it automatically generated somehow..? because that would be awesome...

Reply:
VIM Ultisnip feature :)

Reply:
o man that's nice...i must go look for something in emacs.... so VIM can understand hjson? I've only found json for emacs... not completely the same i think..